app/src/main/java/org/xwiki/android/sync/activities/SignInViewFlipper.kt

```diff
@@ -0,0 +1,242 @@
+import android.text.TextUtils
+import android.view.View
+import android.widget.EditText
+import android.widget.TextView
+import org.xwiki.android.sync.Constants
+import org.xwiki.android.sync.R
+import org.xwiki.android.sync.SERVER_ADDRESS
+import org.xwiki.android.sync.auth.AuthenticatorActivity
+import org.xwiki.android.sync.auth.PARAM_USER_PASS
+import org.xwiki.android.sync.auth.PARAM_USER_SERVER
+import org.xwiki.android.sync.rest.XWikiHttp
+import org.xwiki.android.sync.utils.SharedPrefsUtils
+import org.xwiki.android.sync.utils.getValue
+import rx.Subscription
+import rx.android.schedulers.AndroidSchedulers
+import rx.functions.Action1
+
+/**
+ * Tag for logging.
+ */
+private val TAG = "SignInViewFlipper"
+
+/**
+ * Main auth flipper.
+ *
+ * @version $Id: cc56e12004e2982133afa0c17258b164b8203a15 $
+ */
+class SignInViewFlipper
+/**
+ * Standard constructor
+ *
+ * @param activity Current activity
+ * @param contentRootView Root view of this flipper
+ */
+    (
+    activity: AuthenticatorActivity,
+    contentRootView: View
+) : BaseViewFlipper(activity, contentRootView) {
+
+    /**
+     * Typed username.
+     */
+    private lateinit var accountName: String
+
+    /**
+     * Typed password.
+     */
+    private lateinit var accountPassword: String
+
+    init {
+        findViewById<View>(R.id.signInButton).setOnClickListener {
+            if (checkInput()) {
+                mActivity.showProgress(
+                    mContext.getText(R.string.sign_in_authenticating),
+                    submit()
+                )
+            }
+        }
+    }
+
+    /**
+     * Calling when user push "login".
+     */
+    override fun doNext() {}
+
+    /**
+     * Return to setting server ip address, calling by pressing "back".
+     */
+    override fun doPrevious() {}
+
+    /**
+     * @return true if current input correct and variables [.accountName] and
+     * [.accountPassword] was correctly set
+     */
+    private fun checkInput(): Boolean {
+         return findViewById<EditText>(R.id.accountPassword).let { field ->
+             accountPassword = field.text.toString()
+            field.error = null
+            when {
+                TextUtils.isEmpty(field.text) || field.length() < 5 -> {
+                    field.requestFocus()
+                    field.error = mContext.getString(R.string.error_invalid_password)
+                    false
+                }
+                else -> {
+                    field.error = null
+                    true
+                }
+
+            }
+        } && findViewById<EditText>(R.id.accountName).let { field ->
+             field.error = null
+             accountName = field.text.toString()
+             when {
+                 TextUtils.isEmpty(field.text) -> {
+                     field.requestFocus()
+                     field.error = mContext.getString(R.string.error_field_required)
+                     false
+                 }
+                         else -> {
```

Comment:
must be on one column with TextUtils from previous variant

Reply:
Done